Document/0x05h-Testing-Platform-Interaction.md

```diff
@@ -1381,6 +1381,36 @@ There are several ways to perform dynamic analysis:
 1. For the actual persistence: Use the techniques described in the data storage chapter.
 2. For reflection-based approaches: Use Xposed to hook into the deserialization methods or add unprocessable information to the serialized objects to see how they are handled (e.g., whether the application crashes or extra information can be extracted by enriching the objects).
 
+### Testing for Tapjacking Vulnerability (MSTG-PLATFORM-9)
+
+#### Overview
+
+Tapjacking is a vulnerability in which the malicious application abuses the screen overlay feature of Android to capture user taps and keystrokes. This is possible because an active screen overlay can listen for taps and intercept any information being passed in the underlying activity.
+
+Over the years many known malwares like MazorBot, BankBot, MysteryBot have been abusing screen overlay feature to target business critical applications, namely in the banking sector. The [blog](https://www.infosecurity-magazine.com/opinions/overlay-attacks-safeguard-mobile/ "blog") discusses more about these type of malwares.
+
+#### Static Analysis
+
+Check the source code for implementation of any of the below techniques:
+
+- Override [`onFilterTouchEventForSecurity`](https://developer.android.com/reference/android/view/View#onFilterTouchEventForSecurity%28android.view.MotionEvent%29 "onFilterTouchEventForSecurity")
+- Set [`android:filterTouchesWhenObscured`](https://developer.android.com/reference/android/view/View.html#setFilterTouchesWhenObscured%28boolean%29 "android:filterTouchesWhenObscured") to true
+- [FLAG_WINDOW_IS_OBSCURED](https://developer.android.com/reference/android/view/MotionEvent.html#FLAG_WINDOW_IS_OBSCURED "FLAG_WINDOW_IS_OBSCURED")
+
+When either of the first two controls are implemented, the android framework will discard touches or taps that are received from any other visible window. This will safeguard users against this severe vulnerability.
+
+The third control, when applied, provides safeguard to the components and not to the whole application. This type of mitigation can be used when there is a business need to allow overlay feature by third party application. In this cases the developer must identify sensitive input fields from the application and should associate it with the third control. This would minimise the impact of Tapjacking vulnerability.
+
+Hence based upon the requirement any of these controls can be implemented.
+
+When any of the first two controls is implemented, the applications is not vulnerable anymore to Tapjacking.
+
+#### Dynamic Analysis
+
+To identify the existence of this vulnerability, create a sample APK consisting of a single activity and Toast dialog. Upon installing the sample APK on the device and post launching it, if the toast dialog overlays on the application to test then it is vulnerable to Tapjacking.
```

Comment:
Unfortunately it's not clear how to test this and I agree with @TheDauntless, more details and research is required here.

Reply:
Apologies for the delay @cpholguera 

Thanks @TheDauntless  for suggesting Cloak and Dagger. 

I went through to the research and noted got clarity that,

- Cloak and dagger are basically attacks demonstarting new tricks to abuse overlay functionality. It has series of exploits like ContextAwareClickjacking, Invisible Grid Attacks, etc. 
- The root cause of this is SYSTEM_ALERT_WINDOW permission which is used by overlays. Thus preventing overlay will prevent the application from all these series of exploits. 

Please let me know your views on the same.

For detecting Tapjacking, Apart from QARK, Soon I would be submitting an apk created by me to the mobile security project, which can be leveraged to check this vulnerability.